Autoload extensions

cwillisf opened the pull request:
### Resolves

Resolves #734 

### Proposed Changes

- While deserializing JSON (SB2 or SB3), track the extensions referenced in blocks.
- Before installing the deserialized targets, load any necessary extensions.

Note that in the SB2 case, "the extensions referenced in blocks" are detected by inspecting the converted SB3 opcode. This means that blocks converted from built-in to extension, like pen blocks, can cause the pen extension to be auto-loaded even though there's no record of a pen extension in the SB2's "info" section.

### Reason for Changes

This should allow SB2 and SB3 projects using pen or WeDo 2.0 blocks to work after loading, with no additional steps.

### Test Coverage

Existing tests were altered but no tests were added.

### Known Issues

I suspect this is a pre-existing issue but it's easier to encounter now: deserialized extension block menus always select default value (#757).

Commit:
Load extensions when loading a project or sprite

- Accumulate extension info while deserializing JSON
- Install extensions (if any) before installing target(s)
- Merged 'install' steps of the "add sprite" and "load project" paths

Commit:
Fix tests & a typo

paulkaplan commented:
I believe you should now be able to remove this line:
https://github.com/LLK/scratch-vm/blob/develop/test/integration/pen.js#L51

Commit:
Pen integration test: don't pre-load pen extension

cwillisf commented:
I hear that SB2 WeDo 2.0 extension detection / loading isn't working. Investigating...

paulkaplan commented:
One thing with wedo2 import is that menus get turned into inputs:

![image](https://user-images.githubusercontent.com/654102/32393665-caa75b64-c0b0-11e7-98c0-7b88aac58113.png)

Also trying to import any of the 3 reporter blocks gets messed up.

paulkaplan commented:
(since this is somewhat beyond the scope of autoloading the pen extension which works fine, i'm also ok with filing these as bugs and merging)

Commit:
Actually finish SB2 specMap for WeDo 2.0

Apparently I took a break at the wrong time and forgot to come back to
this... :/

paulkaplan reviewed (approved):
Apart from the code in the sb2specmap that is broken for the wedo2 extension, the rest looks good!
